third_party/move/move-vm/runtime/src/storage/ty_layout_converter.rs

```diff
@@ -77,8 +85,14 @@ where
         gas_meter: &mut impl DependencyGasMeter,
         traversal_context: &mut TraversalContext,
         ty: &Type,
+        check_option_type: bool,
```

Comment:
added this flag to avoid duplicating the code for checking option, any suggestion to make it cleaner?

Reply:
Hopefully we can remove this once feature is enabled